pkg/cmd/roachtest/operation/operation_interface.go

```diff
@@ -0,0 +1,32 @@
+// Copyright 2024 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package operation
+
+import "github.com/cockroachdb/cockroach/pkg/roachprod/logger"
+
+type Operation interface {
+	// ClusterCockroach returns the path to the Cockroach binary on the target
+	// cluster.
+	ClusterCockroach() string
+	Name() string
+	// Spec returns the *registry.OperationSpec as an interface{}.
+	Spec() interface{}
+	Error(args ...interface{})
+	Errorf(string, ...interface{})
+	FailNow()
+	Fatal(args ...interface{})
+	Fatalf(format string, args ...interface{})
+	Failed() bool
+
+	L() *logger.Logger
+	Progress(float64)
```

Comment:
AFAICT, not roachtest currently uses this. I'm not convinced every operation should implement it, especially since the notion of "progress" is most likely undefined in the majority of cases.

Reply:
Removed.